.github/workflows/check_pipeline_status.yml

```diff
@@ -0,0 +1,67 @@
+name: Check Pipeline Status
+
+on:
+  workflow_call:
+    inputs:
+      region: # AWS region where pipeline is located
+        required: true
+        type: string
+      pipeline_name: # Name of the pipeline to check
+        required: true
+        type: string
+      stage_name: # Name of the stage to check
+        required: true
+        type: string
+      retry_attempts: # Number of seconds to wait between retries
+        required: true
+        type: number
+        default: 30
+      retry_interval: # Number of seconds to wait between retries
+        required: true
+        type: number
+        default: 30
+
+# permission can be added at job level or workflow level
+permissions:
+  id-token: write # Required to get the ID Token that will be used for OIDC
+  contents: read # Required for actions/checkout
+
+jobs:
+  check_pipeline_status:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ secrets.AWS_REGION }}
+          role-to-assume: ${{ secrets.BUILD_AWS_IAM_ROLE_ARN }}
+          role-session-name: OIDCSession
+      
+      - name: Check pipeline status
+        run: |
+        
+            retry_counter=0
+
+            while [ $retry_counter -lt ${{ inputs.retry_attempts }}  ] ; do
+
+            if [ $retry_counter -ge ${{ inputs.retry_attempts }} ]; then
+              echo "Pipeline failed after multiple retries"
+              exit 1
+            fi
+
+            STATUS=$(aws codepipeline get-pipeline-state --name ${{ inputs.pipeline_name }} --query "stageStates[?stageName=='${{ inputs.stage_name }}' ].latestExecution.status" --output text)
+            [ $? -ne 0 ] && exit 1 && echo "Error getting pipeline status"
+
+            if [ "$STATUS" == "InProgress" ]; then
+              echo "Pipeline is still in progress"
+              sleep 30
+            elif [ "$STATUS" == "Succeeded" ]; then
+              echo "Pipeline succeeded"
+              exit 0
+            else
+              retry_counter=$[$retry_counter+1]
```

Comment:
Should this be outside of the if statement since it's only incrementing when the status is unknown.

Reply:
@jblackburn22 This is to catch anything but `succeeded` or `inProgress`. For example, this pipeline can be paused or the state changed and will fall into a holding pattern (managed by the interval and retry count) so as to prevent it unnecessarily bailing out. 

If this is removed outside of the condition it will increment every time - which is different behaviour from the `inProgress` check as we assume the inProgress will succeed (although time can vary) - or at least fall into the `else` holding pattern.

Reply:
I have updated the code - based on the other two suggestion so you can see what it looks like.